workflows/raredisease.nf

```diff
@@ -125,16 +123,18 @@ workflow RAREDISEASE {
     // STEP 0: PREPARE GENOME REFERENCES AND INDICES.
     PREPARE_GENOME ( params.fasta )
 
-    // STEP 1: MAPPING READS, FETCH STATS, AND MERGE.
-    MAPPING ( INPUT_CHECK.out.reads, PREPARE_GENOME.out.bwamem2_index )
-    ch_software_versions = ch_software_versions.mix(MAPPING.out.bwamem2_version.ifEmpty(null))
-    ch_software_versions = ch_software_versions.mix(MAPPING.out.picard_version.ifEmpty(null))
-    ch_software_versions = ch_software_versions.mix(MAPPING.out.samtools_version.ifEmpty(null))
+    // STEP 1: ALIGNING READS, FETCH STATS, AND MERGE.
+    if (params.aligner == 'bwamem2') {
+        ALIGN_BWAMEM2 ( INPUT_CHECK.out.reads, PREPARE_GENOME.out.bwamem2_index )
+        ch_software_versions = ch_software_versions.mix(ALIGN_BWAMEM2.out.bwamem2_version.ifEmpty(null))
+        ch_software_versions = ch_software_versions.mix(ALIGN_BWAMEM2.out.picard_version.ifEmpty(null))
+        ch_software_versions = ch_software_versions.mix(ALIGN_BWAMEM2.out.samtools_version.ifEmpty(null))
+    }
```

Comment:
I think we need to name the output channel so that the next step (DV) can be agnostic of the alignment step done before. Similar to [this]( https://github.com/nf-core/rnaseq/blob/8094c42add6dcdf69ce54dfdec957789c37ae903/workflows/rnaseq.nf#L292)